### PR TITLE
Fails in Mir: test for frame timestamps

### LINKED ISSUE #83: Should test surface frame timestamps
Should expose this Mir issue: https://github.com/MirServer/mir/issues/341

### PR DESCRIPTION
**Pull request by wmww**
Fix #83

### CODE REVIEW

File: tests/test_surface_events.cpp [hunk 1]
```diff
@@ -495,6 +495,52 @@ TEST_F(ClientSurfaceEventsTest, surface_moves_while_under_pointer)
         });
 }
 
+TEST_F(ClientSurfaceEventsTest, frame_timestamp_increases)
+{
+    using namespace testing;
+
+    Test::RecordProperty("wlcs-skip-test", "Mir does not send enter events yet");
```
Comment (AlanGriffiths) [inline tests/test_surface_events.cpp]: We should not be implementing Mir specific limitations in these tests.

Any tests that Mir expects to fail ought to be "xfail"ed by the Mir integration code.

Comment (wmww) [reply tests/test_surface_events.cpp]: fixed

File: tests/test_surface_events.cpp [hunk 2]
```diff
@@ -495,6 +495,49 @@ TEST_F(ClientSurfaceEventsTest, surface_moves_while_under_pointer)
...
+    surface.add_frame_callback(
+        [&](int time)
+        {
+            EXPECT_THAT(time, Gt(prev_frame_time));
+            prev_frame_time = time;
+            frame_callback_count++;
+        });
+    wl_surface_commit(surface);
+
+    usleep(10000);
```
Comment (AlanGriffiths) [inline tests/test_surface_events.cpp]: Adding sleeps in the middle of tests smells badly of race conditions.

Why? And isn't there a better way?

**Comment by wmww** [comment]
> I know we have other tests that hang on failure, but let's not follow a bad precedent.

WLCS hanging isn't usually the fault of bad individual tests. It's because `dispatch_until()` doesn't have a timeout, and once that gets fixed everything will stop hanging.

File: tests/test_surface_events.cpp (hunk 2 above)
Comment (wmww) [reply tests/test_surface_events.cpp]: The reason is that we are testing that a millisecond resolution timestamp actually increases as time passes. Therefor we have to wait multiple milliseconds for the test to make sense.

Comment (wmww) [reply tests/test_surface_events.cpp]: The reason is that we are testing that a millisecond resolution timestamp actually increases as time passes. Therefor we have to wait multiple milliseconds for the test to make sense. I will add a comment explaining how it works in more detail.

**Comment by wmww** [comment]
I still don't know how to disable this test in Mir

File: tests/test_surface_events.cpp (hunk 2 above)
Comment (AlanGriffiths) [reply tests/test_surface_events.cpp:534]: It's still a shame we can't simply provide a test timesource run independently of real time.

**Pull request merged**